source/Dialog.cpp

```diff
@@ -224,7 +224,22 @@ bool Dialog::KeyDown(SDL_Keycode key, Uint16 mod, const Command &command, bool i
 {
 	auto it = KEY_MAP.find(key);
 	bool isCloseRequest = key == SDLK_ESCAPE || (key == 'w' && (mod & (KMOD_CTRL | KMOD_GUI)));
-	if((it != KEY_MAP.end() || (key >= ' ' && key <= '~')) && !isMission && (intFun || stringFun) && !isCloseRequest)
+	if(stringFun && (mod & KMOD_CTRL) && (key == 'c' || key == 'v'))
+	{
+		if(key == 'c')
+			SDL_SetClipboardText(input.c_str());
+		else if(SDL_HasClipboardText())
+		{
+			input.clear();
+			char *clipboardText = SDL_GetClipboardText();
+			int n = 0;
+			for(auto cp = clipboardText; *cp && n < 120; cp++, n++)
```

Comment:
Why do we need this 120 char limit?

Reply:
I'm not sure if someone trying to paste more than 120 chars would call this _best practice_.